push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/kafka/AbstractKafkaConsumer.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.kafka;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.jboss.aerogear.unifiedpush.system.ConfigurationUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract Kafka consumer wrapper class that allows its implementations to
+ * simply handle new incoming records.
+ */
+public abstract class AbstractKafkaConsumer implements Runnable {
+
+	private final static Logger logger = LoggerFactory.getLogger(AbstractKafkaConsumer.class);
+
+	/*
+	 * False when a consumer is running; otherwise true
+	 */
+	private final AtomicBoolean closed = new AtomicBoolean(false);
```

Comment:
What is the meaning of "close"? If the variable is false when the consumer is running, wouldn't be better to call it `running` and set it to true?